server/tests/api_v4/authenticated/test_vlob_update.py

```diff
@@ -249,4 +250,218 @@ async def test_authenticated_vlob_update_bad_vlob_version(
+        realm_id=coolorg.wksp1_id,
+        vlob_id=vlob_id,
+        key_index=key_index,
+        blob=b"<initial block content>",
+        timestamp=v1_timestamp,
+        sequester_blob=None,
+    )
+    assert outcome is None, outcome
+    initial_dump = await backend.vlob.test_dump_vlobs(organization_id=coolorg.organization_id)
+
+    v2_timestamp = DateTime.now()
+    rep = await coolorg.alice.vlob_update(
+        vlob_id=vlob_id,
+        key_index=key_index,
+        timestamp=v2_timestamp,
+        version=2,
+        blob=b"<updated block content>",
+        sequester_blob={SequesterServiceID.new(): b"<dummy>"},
+    )
+    assert rep == authenticated_cmds.v4.vlob_update.RepOrganizationNotSequestered()
+
+    # Ensure no changes were made
+    dump = await backend.vlob.test_dump_vlobs(organization_id=coolorg.organization_id)
+    assert dump == initial_dump
+
+
+@pytest.mark.skip(reason="TODO: missing test sequester")
+async def test_authenticated_vlob_update_sequester_inconsistency(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    raise Exception("Not implemented")
+
+
+@pytest.mark.skip(reason="TODO: missing test sequester")
+async def test_authenticated_vlob_update_rejected_by_sequester_service(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    raise Exception("Not implemented")
+
+
+@pytest.mark.skip(reason="TODO: missing test sequester")
+async def test_authenticated_vlob_update_sequester_service_unavailable(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    raise Exception("Not implemented")
+
+
+async def test_authenticated_vlob_update_timestamp_out_of_ballpark(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    initial_dump = await backend.vlob.test_dump_vlobs(organization_id=coolorg.organization_id)
+    t0 = DateTime.now().subtract(seconds=3600)
+    rep = await coolorg.alice.vlob_create(
+        realm_id=coolorg.wksp1_id,
+        vlob_id=VlobID.new(),
+        key_index=1,
+        timestamp=t0,
+        blob=b"<block content>",
+        sequester_blob=None,
+    )
+    assert isinstance(rep, authenticated_cmds.v4.vlob_create.RepTimestampOutOfBallpark)
+    assert rep.ballpark_client_early_offset == 300.0
+    assert rep.ballpark_client_late_offset == 320.0
+    assert rep.client_timestamp == t0
+
+    # Ensure no changes were made
+    dump = await backend.vlob.test_dump_vlobs(organization_id=coolorg.organization_id)
+    assert dump == initial_dump
+
+
+@pytest.mark.parametrize(
+    "timestamp_offset",
+    (pytest.param(0, id="same_timestamp"), pytest.param(1, id="previous_timestamp")),
+)
+async def test_authenticated_vlob_create_require_greater_timestamp(
+    coolorg: CoolorgRpcClients,
+    backend: Backend,
+    timestamp_offset: int,
+) -> None:
+    dt1 = DateTime.now()
+    dt2 = dt1.add(seconds=10)
+    bad_timestamp = dt2.subtract(seconds=timestamp_offset)
+
+    # 1) Create a first vlob
+
+    author = coolorg.alice.device_id
```

Comment:
We can also avoid creating a new vlob and fetch one created for wksp1 (typically the workspace manifest's should already exist and be accessible with `coolorg.wksp1_id`)

Reply:
Done